packages/core-ethereum/src/commitment.ts

```diff
@@ -1,24 +1,28 @@
 import { iterateHash, recoverIteratedHash, HoprDB, Hash } from '@hoprnet/hopr-utils'
 import { randomBytes } from 'crypto'
 import Debug from 'debug'
+import type { Receipt } from './ethereum'
+import Indexer from './indexer'
+import { once } from 'events'
 
 const log = Debug('hopr-core-ethereum:commitment')
 
 export const DB_ITERATION_BLOCK_SIZE = 10000
 export const TOTAL_ITERATIONS = 100000
 
-async function hashFunction(msg: Uint8Array): Promise<Uint8Array> {
+function hashFunction(msg: Uint8Array): Uint8Array {
   return Hash.create(msg).serialize().slice(0, Hash.SIZE)
 }
 
 export class Commitment {
   private initialized: boolean = false
 
   constructor(
-    private setChainCommitment,
-    private getChainCommitment,
+    private setChainCommitment: (commitment: Hash) => Promise<Receipt>,
+    private getChainCommitment: () => Promise<Hash>,
     private db: HoprDB,
-    private channelId: Hash // used in db key
+    private channelId: Hash, // used in db key
+    private indexer: Indexer
```

Comment:
Why does a commitment need an instance of indexer?

Reply:
... it needs to know whether there has been a transaction that sets the on-chain commitment and then delay `Commitment.initialize()` until the updated state appears in the indexer

Reply:
This seems like bad design. Why not a dedicated function passed in for this purpose?

Reply:
Sounds good, changing